chainer/training/updaters/multiprocess_parallel_updater.py

```diff
@@ -265,6 +265,9 @@ def finalize(self):
         for worker in self._workers:
             worker.join()
 
+        for iterator in six.itervalues(self._iterators):
+            iterator.finalize()
```

Comment:
A suggestion but you can also call super's finalize. Not sure which is cleaner.

Reply:
Yes, I was also thinking on that, I just adapted it. Thanks!!